src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentDesignTimeNodeWriter.cs

```diff
@@ -275,26 +300,51 @@ public override void WriteHtmlContent(CodeRenderingContext context, HtmlContentI
             // Do nothing
         }
 
-        protected override void BeginWriteAttribute(CodeWriter codeWriter, string key)
+        protected override void BeginWriteAttribute(CodeRenderingContext context, string key)
         {
-            if (codeWriter == null)
+            if (context == null)
             {
-                throw new ArgumentNullException(nameof(codeWriter));
+                throw new ArgumentNullException(nameof(context));
             }
 
             if (key == null)
             {
                 throw new ArgumentNullException(nameof(key));
             }
 
-            codeWriter
+            context.CodeWriter
                 .WriteStartMethodInvocation($"{_scopeStack.BuilderVarName}.{nameof(ComponentsApi.RenderTreeBuilder.AddAttribute)}")
                 .Write("-1")
                 .WriteParameterSeparator()
                 .WriteStringLiteral(key)
                 .WriteParameterSeparator();
         }
 
+        protected override void BeginWriteAttribute(CodeRenderingContext context, IntermediateNode expression)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            if (expression == null)
+            {
+                throw new ArgumentNullException(nameof(expression));
+            }
+
+            context.CodeWriter.WriteStartMethodInvocation($"{_scopeStack.BuilderVarName}.{ComponentsApi.RenderTreeBuilder.AddAttribute}");
+            context.CodeWriter.Write("-1");
```

Comment:
I know this was already there but what is this -1 for?

Reply:
Is it because the sequence doesn't matter in design time?

Reply:
yes, our design-time codegen is simplified. Since we have to have two versions, the design-time version is comparatively stripped down.